### ISSUE TITLE
[Imprv] add functionality constructed to module inventory

### ISSUE THREAD

**Issue by peshay** [nutanix/nutanix.ansible#235]
**Describe the request**
Enable [Constructed Features](https://docs.ansible.com/ansible/latest/dev_guide/developing_inventory.html#constructed-features) for the inventory plugin

**Current behaviour**
There is no way to create additional groups from the inventory plugin.

**Expected behaviour**
Add constructed functionality to the plugin, so it is possible to create additional vars and groups based on Jinja2.

For example
```yaml
plugin: nutanix.ncp.ntnx_prism_vm_inventory
groups:
  nutanix: true
keyed_groups:
  # group for customer name
  - prefix: ""
    separator: ''
    key: "inventory_hostname | regex_replace('([a-zA-Z0-9]+)-([a-zA-Z0-9]+)-([a-zA-Z0-9]+)', '\\1')"
  # group for service name
  - prefix: ""
    separator: ''
    key: "inventory_hostname | regex_replace('([a-zA-Z0-9]+)-([a-zA-Z0-9]+)-([a-zA-Z]+)[0-9]*', '\\3')"
  # group for environemnt name
  - prefix: ""
    separator: ''
    key: "inventory_hostname | regex_replace('([a-zA-Z0-9]+)-([a-zA-Z0-9]+)-([a-zA-Z0-9]+)', '\\2')"
  # group for customer environemnt name
  - prefix: ""
    separator: ''
    key: "inventory_hostname | regex_replace('([a-zA-Z0-9]+)-([a-zA-Z0-9]+)-([a-zA-Z0-9]+)', '\\1_\\2')"
```

Here is a pull request to add the requested functionality: https://github.com/nutanix/nutanix.ansible/issues/232

**Comment by bhati-pradeep** [comment]
@peshay I am looking into it. One use-case enhancement we need to add more here is the grouping using categories of vm.

**Comment by bhati-pradeep** [comment]
Released in v1.7.0
